### ISSUE TITLE
AWS Lambda for China region(cn-norh-1) is available now

### ISSUE THREAD

**Issue by TerrificMao** [serverless/serverless#4463]
AWS Lambda for China region(cn-norh-1) is available now, could you please support it?  

Serverless Error ---------------------------------------
  The security token included in the request is invalid.
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           7.6.0
     Serverless Version:     1.20.2

**Comment by RafalWilinski** [comment]
Hi @TerrificMao, thanks for adding this! 🏅 

I've tried to find any information about this release but couldn't find anything. According to this: https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/ Lambda and APIGW is still not available in Beijing.

Could you please share a blogpost/article related to this? I'd love to read about this.

**Comment by horike37** [comment]
I found the following tweet. Not for sure, you can already use Lambda in China region.
https://twitter.com/david_packman/status/928806476876980225

Generally speaking, Serverless doesn't need any updates for supporting a new Region(But China Region is special a little bit, so would need any updates though).
You should see working fine if CloudFormation support for related resources like Lambda, S3 etc...

**Comment by leonli** [comment]
Hi @RafalWilinski I'm sure Lambda & API GW are launched in China Region, and a Chinese blog post has been published, FYR the attached is the console of China Region as you can see the services are out there.
<img width="944" alt="screen shot 2017-11-13 at 4 58 19 pm" src="https://user-images.githubusercontent.com/345188/32717261-e54721c6-c893-11e7-98b8-1188692d05a2.png">

**Comment by menway** [comment]
Dear all fans of AWS China, I also met this error today and following are my experiments:

First of all, add China region setting in your serverless.yml: 

```
provider:
  name: aws
  region: cn-north-1
  runtime: nodejs6.10
```

And then the security token error should be fixed. (I guess AWS sdk will correctly send requests to AWS China server if you setup `cn-north-1` as the region, otherwise it will send requests to AWS Global server)

But when you continue, you will meet the following error:
```
S3 error: The specified bucket does not exist
```

This was becaused there are some places in serverless source code that hard coded AWS S3 domain to `s3.amazonaws.com` (this is for AWS Global, not correct for China region), so I changed all to `s3.cn-north-1.amazonaws.com.cn` as a temporary workaround.

Then the deploy process could continue, but at last I was blocked by the following error:
```
An error occurred: ApiGatewayRestApi - Resource is not supported in this region.
```

Seems `ApiGatewayRestApi` has not been implemented yet in China region? Any ideas?

**Comment by menway** [comment]
Full logs is pasted here:
```
> SLS_DEBUG=1 serverless deploy --verbose
Serverless: Load command run
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command emit
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Invoke deploy
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:deploy:deploy
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Uploading service .zip file to S3 (404 B)...
Serverless: Validating template...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
CloudFormation - UPDATE_IN_PROGRESS - AWS::CloudFormation::Stack - serverless-hello-world-dev
CloudFormation - CREATE_IN_PROGRESS - AWS::ApiGateway::RestApi - ApiGatewayRestApi
CloudFormation - CREATE_IN_PROGRESS - AWS::IAM::Role - IamRoleLambdaExecution
CloudFormation - CREATE_IN_PROGRESS - AWS::Logs::LogGroup - HelloWorldLogGroup
CloudFormation - CREATE_FAILED - AWS::ApiGateway::RestApi - ApiGatewayRestApi
CloudFormation - CREATE_IN_PROGRESS - AWS::Logs::LogGroup - HelloWorldLogGroup
CloudFormation - CREATE_IN_PROGRESS - AWS::IAM::Role - IamRoleLambdaExecution
CloudFormation - CREATE_COMPLETE - AWS::Logs::LogGroup - HelloWorldLogGroup
CloudFormation - CREATE_FAILED - AWS::IAM::Role - IamRoleLambdaExecution
CloudFormation - UPDATE_ROLLBACK_IN_PROGRESS - AWS::CloudFormation::Stack - serverless-hello-world-dev
CloudFormation - UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS - AWS::CloudFormation::Stack - serverless-hello-world-dev
CloudFormation - DELETE_COMPLETE - AWS::ApiGateway::RestApi - ApiGatewayRestApi
CloudFormation - DELETE_IN_PROGRESS - AWS::IAM::Role - IamRoleLambdaExecution
CloudFormation - DELETE_IN_PROGRESS - AWS::Logs::LogGroup - HelloWorldLogGroup
CloudFormation - DELETE_COMPLETE - AWS::Logs::LogGroup - HelloWorldLogGroup
CloudFormation - DELETE_COMPLETE - AWS::IAM::Role - IamRoleLambdaExecution
CloudFormation - UPDATE_ROLLBACK_COMPLETE - AWS::CloudFormation::Stack - serverless-hello-world-dev
Serverless: Operation failed!

  Serverless Error ---------------------------------------

  An error occurred: ApiGatewayRestApi - Resource is not supported in this region.

  Stack Trace --------------------------------------------

ServerlessError: An error occurred: ApiGatewayRestApi - Resource is not supported in this region.
    at provider.request.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:114:33)
From previous event:
    at AwsDeploy.monitorStack (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:26:12)
    at provider.request.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:88:30)
From previous event:
    at AwsDeploy.update (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:88:8)
From previous event:
    at AwsDeploy.BbPromise.bind.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:105:12)
From previous event:
    at AwsDeploy.updateStack (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:99:8)
From previous event:
    at AwsDeploy.BbPromise.bind.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:125:39)
From previous event:
    at BbPromise.reduce (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:366:55)
From previous event:
    at PluginManager.invoke (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:366:22)
    at PluginManager.spawn (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:384:17)
    at AwsDeploy.BbPromise.bind.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:91:48)
From previous event:
    at Object.deploy:deploy [as hook] (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:87:10)
    at BbPromise.reduce (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:366:55)
From previous event:
    at PluginManager.invoke (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:366:22)
    at PluginManager.run (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/classes/PluginManager.js:397:17)
    at variables.populateService.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/Serverless.js:104:33)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Serverless.run (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/lib/Serverless.js:91:74)
    at serverless.init.then (/home/ubuntu/.nvm/versions/node/v6.11.4/lib/node_modules/serverless/bin/serverless:42:50)

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless

  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           6.11.4
     Serverless Version:     1.24.1
```

**Comment by RafalWilinski** [comment]
Seems like `ApiGatewayRestApi` is not supported yet which is a blocker, at least from my perspective.

**Comment by iwater** [comment]
without http defined in serverless.yml, expected deploy lambda successful without endpoints by api gateway, but there is no lambda found in aws console
`Serverless: Load command run
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command emit
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Invoke deploy
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:deploy:deploy
Serverless: Service files not changed. Skipping deployment...
Serverless: Invoke aws:info
Service Information
service: my-service
stage: dev
region: cn-north-1
stack: my-service-dev
api keys:
  None
endpoints:
  None
functions:
  hello: my-service-dev-hello

Stack Outputs
ServerlessDeploymentBucketName: my-service-dev-serverlessdeploymentbucket-qicyadupz8i6

Serverless: Publish service to Serverless Platform...
Service successfully published! Your service details are available at:
https://platform.serverless.com/services/iwater/my-service
Serverless: Invoke aws:deploy:finalize`

**Comment by horike37** [comment]
just opened up [another issue](#4615) for supporting cn-north region so closing this issue for now.
